File: hw/gudino/u3/hw29Singleton/SingletonPython/model/inventory.py
from model.product import Product

class Inventory:
    def __init__(self):
        self._products = []

    def get_products(self):
        return self._products

    def add_product(self, product: Product):
        self._products.append(product)

    def sell_product(self, index: int, quantity: int):
        self._products[index].sell(quantity)

    def restock_product(self, index: int, quantity: int):
        self._products[index].restock(quantity)

File: hw/gudino/u3/hw29Singleton/SingletonJS/model/Inventory.js
export default class Inventory {
    constructor() {
        this.products = [];
    }

    getProducts() {
        return this.products;
    }

    addProduct(product) {
        this.products.push(product);
    }

    sellProduct(index, quantity) {
        this.products[index].sell(quantity);
    }

    restockProduct(index, quantity) {
        this.products[index].restock(quantity);
    }
}
